src/libraries/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/CanonicalXmlDocument.cs

```diff
@@ -61,54 +61,54 @@ public void WriteHash(HashAlgorithm hash, DocPosition docPos, AncestralNamespace
             }
         }
 
-        public override XmlElement CreateElement(string prefix, string localName, string namespaceURI)
+        public override XmlElement CreateElement(string? prefix, string localName, string? namespaceURI)
         {
-            return new CanonicalXmlElement(prefix, localName, namespaceURI, this, _defaultNodeSetInclusionState);
+            return new CanonicalXmlElement(prefix!, localName, namespaceURI!, this, _defaultNodeSetInclusionState);
```

Comment:
Here `prefix` shouldn't be nullable

Reply:
This is another of those situations where this method overrides the base method and so has to have the same signature. The nullable parameters here are incompatible with the constructor, which disallows nulls. It disallows nulls because the constructor calls straight into `base` which also disallows nulls.

I can't see a way to override the, er, override, and specify different annotations.

Reply:
Hm, yes, too late to change annotations in `XmlDocument`, but why `namespaceURI` is non-nullable in `CanonicalXmlElement`? It is nullable in `XmlElement`'s ctor (where you are passing it to)

Reply:
All of the callers to `CreateElement` pass a non-null `string`, so I chose the overload that takes a non-nullable string. The nullability in this parameter is only here because it needs to be for the overload, but in effect, the parameter is never null.